### ISSUE TITLE
Read-only files not deletable on Windows

### ISSUE THREAD

**Issue by avdv** [syncthing/syncthing#1610]
I was syncing some files from a Linux machine to the Windows 8 host, then deleted some of those files again.

The log on the Windows machine turned up these errors:

```
[GUEL3] 22:27:05 INFO: Puller (folder "Photos", file "2012\\08\\04\\124620.jpeg")
: delete: remove C:\Users\Fred\Photos\2012\08\04\124620.jpeg: access denied
[GUEL3] 22:27:05 INFO: Puller (folder "Photos", file "2012\\08\\04\\124628.jpeg")
: delete: remove C:\Users\Fred\Photos\2012\08\04\124628.jpeg: access denied
[GUEL3] 22:27:05 WARNING: Folder "Photos" isn't making progress - check logs for
possible root cause. Pausing puller for 1m0s.
```
I could delete those files manually using the Windows explorer, but trying to run del in a cmd window gave an error too:

```
> del C:\Users\Fred\Photos\2012\08\04\124628.jpeg
C:\Users\Fred\Photos\2012\08\04\124628.jpeg
Access is denied.
```

Turns out, that on Windows, you cannot simply delete files having the `READONLY` flag set. See the discussion here: https://github.com/golang/go/issues/9606

Basically, you would have to remove the `read-only` attribute prior to trying call os.remove.

**Comment by AudriusButkevicius** [comment]
Right, and what's the bug?

**Comment by AudriusButkevicius** [comment]
Actually, I now understand.
It's weird that this actually happens.

**Comment by meztup** [comment]
This issue still occurs for **folders** with the read-only attribute set. This is usually not an issue, but Windows sets this attribute on folders automatically when you set a custom icon for said folder. In those cases, a delete of the folder on one host results in "out of sync" on the other until one manually deletes the folder on the "out of sync" side.
